src/main/java/seedu/address/logic/parser/FindBuyerCommandParser.java

```diff
@@ -0,0 +1,55 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import java.util.function.Predicate;
+
+import seedu.address.logic.commands.FindCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.Buyer;
+import seedu.address.model.person.Deliverer;
+import seedu.address.model.person.PersonCategory;
+import seedu.address.model.person.Supplier;
+
+/**
+ * Parses input arguments and creates a new FindCommand object.
+ */
+public class FindBuyerCommandParser implements Parser<FindCommand> {
+    public static final String PARSE_WORD = "find-b";
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the FindCommand
+     * and returns a FindCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format.
+     */
+    public FindCommand parse(String args) throws ParseException {
+        String trimmedArgs = args.trim();
+        if (trimmedArgs.isEmpty()) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        }
+
+        Predicate<Buyer> bPredicate = PredicateParser.parseBuyer(trimmedArgs);
```

Comment:
Maybe it will be clearer in meaning to change `bPredicate` to `buyerPredicate`. The same applies to below.